editing/edit-with-branch-versioning/src/main/java/com/esri/samples/edit_with_branch_versioning/EditWithBranchVersioningController.java

```diff
@@ -0,0 +1,359 @@
+  private ServiceGeodatabase serviceGeodatabase; // keeps loadable in scope to avoid garbage collection
+  private String defaultVersion;
+  private String userCreatedVersion;
+
+  public void initialize() {
+    try {
+      // create a map with the streets vector basemap and set it to the map view
+      ArcGISMap map = new ArcGISMap(Basemap.createStreetsVector());
+      mapView.setMap(map);
+
+      // add the version access types to the access type combo box
+      accessTypeComboBox.getItems().addAll(VersionAccess.PUBLIC, VersionAccess.PROTECTED, VersionAccess.PRIVATE);
+
+      // add the damage types to the damage type combo box
+      damageTypeComboBox.getItems().addAll("Destroyed", "Inaccessible", "Major", "Minor", "Affected");
+      // add a listener to handle damage type attribute selections and update the selected feature with the new value
+      damageTypeComboBox.getSelectionModel().selectedItemProperty().addListener((o, p, n) -> {
+        if (selectedFeature != null && !selectedFeature.getAttributes().get("TYPDAMAGE").equals(n)) {
+          selectedFeature.getAttributes().put("TYPDAMAGE", damageTypeComboBox.getValue());
+          updateFeature(selectedFeature);
+        }
+      });
+
+      // handle authentication for the service geodatabase
+      AuthenticationManager.setAuthenticationChallengeHandler(new DefaultAuthenticationChallengeHandler());
+
+      // create and load a service geodatabase
+      serviceGeodatabase = new ServiceGeodatabase("https://sampleserver7.arcgisonline" +
+        ".com/arcgis/rest/services/DamageAssessment/FeatureServer");
+      serviceGeodatabase.loadAsync();
+      serviceGeodatabase.addDoneLoadingListener(() -> {
+        if (serviceGeodatabase.getLoadStatus() == LoadStatus.LOADED) {
+
+          // when the service geodatabase has loaded get the default version
+          defaultVersion = serviceGeodatabase.getDefaultVersionName();
+
+          // get the service feature table from the service geodatabase
+          if (serviceGeodatabase.getTable(0) != null) {
+            serviceFeatureTable = serviceGeodatabase.getTable(0);
+
+            // create a feature layer from the service feature table and add it to the map
+            featureLayer = new FeatureLayer(serviceFeatureTable);
+            map.getOperationalLayers().add(featureLayer);
+            featureLayer.addDoneLoadingListener(() -> {
+              if (featureLayer.getLoadStatus() == LoadStatus.LOADED) {
+
+                // when the feature layer has loaded set the viewpoint and update the UI
+                mapView.setViewpointAsync(new Viewpoint(featureLayer.getFullExtent()));
+                progressIndicator.setVisible(false);
+                createVersionButton.setDisable(false);
+                currentVersionLabel.setText("Current version: " + serviceGeodatabase.getVersionName());
+              } else {
+                new Alert(Alert.AlertType.ERROR, "Feature layer failed to load").show();
```

Comment:
just a wee tip, can ignore or implement as you prefer! 

when a one liner follows an else block, you can move all the code on to one line for improved readability:

```suggestion
              } else new Alert(Alert.AlertType.ERROR, "Feature layer failed to load").show();
```

Reply:
Thanks, I think that will help in this sample as there's quite a few of them so tidies it up a bit!